test/util/IdTableHelpers.cpp

```diff
@@ -0,0 +1,207 @@
+  auto trace{generateLocationTrace(l, traceMessage.str())};
+
+  // Because we compare tables later by sorting them, so that every table has
+  // one definit form, we need to create local copies.
+  IdTable localTable{table.clone()};
+  IdTable localExpectedContent{expectedContent.clone()};
+
+  if (resultMustBeSortedByJoinColumn) {
+    // Is the table sorted by join column?
+    ASSERT_TRUE(std::ranges::is_sorted(localTable.getColumn(joinColumn)));
+  }
+
+  // Sort both the table and the expectedContent, so that both have a definite
+  // form for comparison.
+  std::ranges::sort(localTable, std::ranges::lexicographical_compare);
+  std::ranges::sort(localExpectedContent, std::ranges::lexicographical_compare);
+
+  ASSERT_EQ(localTable, localExpectedContent);
+}
+
+// ____________________________________________________________________________
+void sortIdTableByJoinColumnInPlace(IdTableAndJoinColumn& table) {
+  CALL_FIXED_SIZE((std::array{table.idTable.numColumns()}), &Engine::sort,
+                  &table.idTable, table.joinColumn);
+}
+
+// ____________________________________________________________________________
+IdTable generateIdTable(
+    const size_t numberRows, const size_t numberColumns,
+    const std::function<IdTable::row_type()>& rowGenerator) {
+  AD_CONTRACT_CHECK(numberRows > 0 && numberColumns > 0);
+
+  // Creating the table and setting it to the wanted size.
+  IdTable table{numberColumns, ad_utility::testing::makeAllocator()};
+  table.resize(numberRows);
+
+  // Fill the table.
+  std::ranges::generate(table, [&numberColumns, &rowGenerator]() {
+    // Make sure, that the generated row has the right size, before passing it
+    // on.
+    IdTable::row_type row = rowGenerator();
+
+    if (row.numColumns() != numberColumns) {
+      throw std::runtime_error(absl::StrCat(
+          "The row generator generated a IdTable::row_type of size ",
+          row.numColumns(), ", which is not the wanted size ", numberColumns,
+          "."));
+    }
+
+    return row;
+  });
+
+  return table;
+}
+
+// ____________________________________________________________________________
+IdTable createRandomlyFilledIdTable(
+    const size_t numberRows, const size_t numberColumns,
+    const std::vector<std::pair<size_t, std::function<ValueId()>>>&
+        joinColumnWithGenerator) {
+  AD_CONTRACT_CHECK(numberRows > 0 && numberColumns > 0);
+
+  // Views for clearer access.
+  auto joinColumnNumberView = std::views::keys(joinColumnWithGenerator);
+  auto joinColumnGeneratorView = std::views::values(joinColumnWithGenerator);
+
+  // Are all the join column numbers within the max column number?
+  if (auto column = std::ranges::find_if(
+          joinColumnNumberView,
+          [&numberColumns](const size_t& num) { return num >= numberColumns; });
+      column != joinColumnNumberView.end()) {
+    throw std::runtime_error(
+        absl::StrCat("The join columns", *column, " is out of bounds."));
```

Comment:
I think for those internal utils, you can shorten these errors, and just use a 
`AD_CONTRACT_CHECK(ranges::all_of)`

Reply:
Done, see [c6acf18](https://github.com/ad-freiburg/qlever/pull/1016/commits/c6acf1848569130ede2a91d66fae78b28d5118c6).